src/main/java/main/Parser.java

```diff
@@ -0,0 +1,201 @@
+            int locationOfTo = 0;
+            for (int i = 0; i < splitInput.length; i++) {
+                if (splitInput[i].equals("/from")) {
+                    locationOfFrom = i;
+                }
+                if (splitInput[i].equals("/to")) {
+                    locationOfTo = i;
+                }
+            }
+            StringBuilder inputAfterFrom = new StringBuilder();
+            StringBuilder inputAfterTo = new StringBuilder();
+            for (int i = locationOfFrom + 1; i < locationOfTo; i++) {
+                inputAfterFrom.append(splitInput[i] + " ");
+            }
+            for (int i = locationOfTo + 1; i < splitInput.length; i++) {
+                inputAfterTo.append(splitInput[i] + " ");
+            }
+            String finalInput = inputAfterFrom.toString().trim();
+            this.checkValidDateFormat(finalInput, "☹ OOPS!!! The even time cannot be empty.");
+            this.checkValidDateFormat(finalInput, "☹ OOPS!!! The deadline cannot be empty.");
+            LocalDateTime dateFrom = parseDate(finalInput);
+            LocalDateTime dateTo = parseDate(finalInput);
+            int startIndex = command.length();
+            String taskName = input.substring(startIndex).split("/")[0];
+            Task newTask = new Events(taskName, dateFrom, dateTo);
+            return new EventsCommand(newTask);
+        } else if (command.equals("find")) {
+            checkKeywordFormat(splitInput);
+            return new FindCommand(splitInput[1]);
+        } else {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    private void validateInputNotEmpty(String[] splitInput, String message, int noOfInputs) throws DukeException {
+        if (splitInput.length < INPUT_LENGTH_VALIDATE) {
+            throw new DukeException(message);
+        }
+    }
+
+    private void checkValidDateFormat(String input, String message) throws DukeException {
+        String[] splitInput = input.split("/");
+        if (splitInput.length != 3 || !isNumeric(splitInput[0]) || !isNumeric(splitInput[1])) {
+            throw new DukeException("Not valid date format. Format must be dd/mm/yyyy HHmm");
+        }
+
+        String[] yearAndTime = splitInput[2].split(" ");
+        if (!(yearAndTime.length == 2 && isNumeric(yearAndTime[0]) && isNumeric(yearAndTime[1]))) {
+            throw new DukeException("Not a valid date format Format must be dd/mm/yyyy HHmm");
+        }
+    }
+
+    private void checkKeywordFormat(String[] splitInput) throws DukeException {
+        if (splitInput.length > INPUT_LENGTH_KEYWORD) {
+            throw new DukeException("Cannot have multiple keywords sorry :(.");
+        }
+    }
+
+    /**
+     * Check whether the string is a real number.
+     * 
+     * @param string any string
+     * @return true if it's a number false otherwise
+     */
+    public static boolean isNumeric(String string) {
```

Comment:
Perhaps naming this as a verb as per our naming conventions would be better? e.g. `numericCheck()`